browser/ui/views/speedreader/speedreader_bubble_single_page.h

```diff
@@ -0,0 +1,74 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_UI_VIEWS_SPEEDREADER_SPEEDREADER_BUBBLE_SINGLE_PAGE_H_
+#define BRAVE_BROWSER_UI_VIEWS_SPEEDREADER_SPEEDREADER_BUBBLE_SINGLE_PAGE_H_
+
+#include "brave/browser/ui/speedreader/speedreader_bubble_view.h"
+#include "chrome/browser/ui/views/location_bar/location_bar_bubble_delegate_view.h"
+#include "ui/views/metadata/metadata_header_macros.h"
+
+namespace content {
+class WebContents;
+}  // namespace content
+
+namespace ui {
+class Event;
+}  // namespace ui
+
+namespace views {
+class View;
+class Label;
+class StyledLabel;
+}  // namespace views
+
+namespace speedreader {
+
+class ReaderButton;
+class SpeedreaderTabHelper;
+
+// SpeedreaderBubbleSinglePage is the bubble shown when the user has enabled
+// reader mode but has not enabled Speedreader globally
```

Comment:
I'm probably missing something: what is the difference between enabled reader mode and speedreader enabled globally?

Reply:
It’s the design spec for how we introduce users to speedreader

Reply:
If the page is readable, there is a little book icon that pops up in the address bar, like other reader modes. The user can distill just that page. If Speedreader is not enabled, SpeedreaderSinglePageBubble pops up, prompting the user to enable speedreader.

In the case that speedreader is enabled, if they click the bubble on any site the SpeedreaderBubblGlobal pops up.

Reply:
i modified the comments to be a bit more clear.